agent/go.mod

```diff
@@ -0,0 +1,114 @@
+module github.com/pulumi/pulumi-kubernetes-operator/agent
```

Comment:
Structure nits:

* We should put the new code under a v2 subdirectory.
* We should try as hard as possible to keep one `go.mod` at the root of the repo, unless there's a legitimate version incompatibility that's non-trivial to resolve between v1 and v2. (Is there?)
* If we absolutely need a new module, it should be under v2 so the agent and server both belong to it since they're versioned together.

Unnecessary modules are a big source of accidental complexity and lead to all sorts of headaches and gotchas down the line ([example](https://github.com/pulumi/pulumi-service/pull/20778)). Trust me on this one :)

Reply:
I plan to tackle the larger structural stuff in a follow-up.